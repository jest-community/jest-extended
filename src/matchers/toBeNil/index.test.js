import each from 'jest-each';

import matcher from './';

expect.extend(matcher);

describe('.toBeNil', () => {
  it('passes when null is given', () => {
    expect(null).toBeNil();
  });

  it('passes when undefined is given', () => {
    expect(undefined).toBeNil();
  });
  it('fails when the value is not null or undefined', () => {
    expect(() => expect('value').toBeNil()).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeNil', () => {
  each([['true'], [{}], [true]]).it('passes when value is not null or undefined : %s', given => {
    expect(given).not.toBeNil();
  });

  it('fails when null is given', () => {
    expect(() => expect(null).not.toBeNil()).toThrowErrorMatchingSnapshot();
  });
});
