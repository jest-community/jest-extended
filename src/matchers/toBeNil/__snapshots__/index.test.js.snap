// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeNil fails when null is given 1`] = `
"<dim>expect(</><red>received</><dim>).not.toBeNil(</><dim>)</>

Expected value not to be null or undefined, received:
  <red>null</>"
`;

exports[`.toBeNil fails when the value is not null or undefined 1`] = `
"<dim>expect(</><red>received</><dim>).toBeNil(</><dim>)</>

Expected value to be null or undefined, received:
  <red>\\"value\\"</>"
`;
